rasa/core/policies/rule_policy.py

```diff
@@ -519,6 +519,24 @@ def _handling_loop_sources(domain: Domain) -> Set[Text]:
             )
         return loop_sources
 
+    def _should_delete(
+        self, prediction_source: Text, tracker: TrackerWithCachedStates
+    ) -> bool:
+        """Checks whether this contradiction is due to action, intent pair.
+
+        Args:
+            prediction_source: the states that result in the prediction
+            tracker: the tracker that raises the contradiction
+
+        Returns:
+            true if the contradiction is a result of an action, intent pair in the rule.
+        """
+        if tracker.is_rule_tracker or prediction_source.count(PREVIOUS_ACTION) > 1:
+            return False
+        for source in self.lookup[RULES]:
+            if prediction_source[:-2] in source and not prediction_source == source:
```

Comment:
I think it's better to do
```suggestion
            if source.startswith(prediction_source[:-2]) and not prediction_source == source:
```
this way, we ensure, that it is the first dialogue step